crates/sui-bridge/src/sui_client.rs

```diff
@@ -224,6 +255,52 @@ impl SuiClientInner for SuiSdkClient {
     async fn get_bridge_committee(&self) -> Result<BridgeCommittee, Self::Error> {
         unimplemented!()
     }
+
+    async fn execute_transaction_block_with_effects(
+        &self,
+        tx: Transaction,
+    ) -> Result<SuiTransactionBlockResponse, BridgeError> {
+        match self.quorum_driver_api().execute_transaction_block(
+            tx,
+            SuiTransactionBlockResponseOptions::new().with_effects(),
+            Some(sui_types::quorum_driver_types::ExecuteTransactionRequestType::WaitForEffectsCert),
+        ).await {
+            Ok(response) => Ok(response),
+            Err(sui_sdk::error::Error::UserInputError(err)) => {
+                if matches!(err, UserInputError::ObjectVersionUnavailableForConsumption{..}) {
```

Comment:
It's a bit hard to reason about whether we have covered all the interesting cases here. I think we need something simpler

Reply:
If we follow what you suggested then I think the categorization is not necessary, at least in the beginning